idm/meta/dao/sql/ns-sql.go

```diff
@@ -188,3 +191,29 @@ func (s *nsSqlImpl) List(ctx context.Context) (map[string]*idm.UserMetaNamespace
 
 	return res, nil
 }
+
+func (s *nsSqlImpl) GetJS(ctx context.Context) (*structpb.Struct, error) {
+	var mm []*MetaNamespace
+	tx := s.Session(ctx).Find(&mm).Where("definition IS NOT NULL AND definition != ''")
+	if tx.Error != nil {
+		return nil, nsTag(tx.Error)
+
+	}
+	if len(mm) > 0 {
+		nss := make([]json_schema.NamespaceDescriptor, 0, len(mm))
+		for _, m := range mm {
+			nss = append(nss, json_schema.NamespaceDescriptor{
+				Label:      m.Label,
+				Definition: m.Definition,
+			})
+		}
+
+		schema, err := json_schema.BuildNamespacesJsonSchema(nss)
+		if err != nil {
+			return nil, err
+		}
+		return schema, nil
+	} else {
+		return nil, nil
+	}
```

Comment:
When using a guard clause, leave the success statement for later. So then you can avoid the complex branches and it's much easier to read. You just look to the end and you see the success clause.

Reference https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html
```suggestion
	if len(mm) = 0 {
    return nil, nil
	}

  nss := make([]json_schema.NamespaceDescriptor, 0, len(mm))
  for _, m := range mm {
    nss = append(nss, json_schema.NamespaceDescriptor{
      Label:      m.Label,
      Definition: m.Definition,
    })
  }

  schema, err := json_schema.BuildNamespacesJsonSchema(nss)
  if err != nil {
    return nil, err
  }

  return schema, nil
```

Reply:
I planned to revisit this part soon, but this makes sense.